broker-daemon/block-order-worker/index.js

```diff
@@ -101,6 +101,24 @@ class BlockOrderWorker extends EventEmitter {
 
     const blockOrder = new BlockOrder({ id, marketName, side, amount, price, timeInForce })
 
+    // If the user tries to place an order for more than they hold in the counter engine channel, throw an error
+    if (blockOrder.side === BlockOrder.SIDES['BID']) {
+      const counterEngine = this.engines.get(blockOrder.counterSymbol)
+      const amountInChannel = await counterEngine.getTotalChannelBalance()
```

Comment:
don't we have an engine method that specifically checks if there are sufficient funds?